shortcode.php

```diff
@@ -330,3 +330,168 @@ function geoip_detect_shortcode_user_info($attr) {
+ * "timezone", "continent", "country", "region"/"state", "city", and "and_not".
+ *
+ * Each attribute may only appear once in a shortcode, and each attribute can only contain a single value, or
+ * the shortcode will break, showing the contents no matter what.
+ *
+ * Examples:
+ *
+ * `[geoip_detect2_show_if country="US" state="TX"]<h1>Title</h1>[/geoip_detect2_show_if]`
+ * 	    - OR -
+ * `[geoip_detect2_show_if country="US" region="TX"]<h1>Title</h1>[/geoip_detect2_show_if]`
+ * 	    - OR -
+ * `[geoip_detect2_show_if country="US" region="Texas"]<h1>Title</h1>[/geoip_detect2_show_if]`
+ * Only displays the inner content if the specified attributes match the visitor's country and region/state.
+ *
+ * `[geoip_detect2_show_if country="US" state="TX" and_not="Houston"]<h1>Title</h1>[/geoip_detect2_show_if]`
+ * Only displays the inner content if the specified attributes match the visitor's country and region/state,
+ * and the specified exclusion is not found in any possible GeoIP Detect field (e.g. visitor's country,
+ * city, timezone, etc.).
+ *
+ * `[geoip_detect2_show_if continent="North America"]<h1>Title</h1>[/geoip_detect2_show_if]`
+ * Only displays the inner content if the specified attributes match the visitor's continent.
+ *
+ */
+function geoip_detect2_shortcode_show_if($atts, $content = null, $shortcodeName = '') {
+    $expectedResult = ($shortcodeName == 'geoip_detect2_show_if') ? true : false;
+
+    $atts_array = shortcode_atts(array(
+        'timezone' => null,
+        'continent' => null,
+        'country' => null,
+        'most_specific_subdivision' => null,
+        'region' => null,
+        'state' => null,
+        'city' => null,
+        'and_not' => null,
+        'not_country' => null),
+        $atts);
+
+    $info = geoip_detect2_get_info_from_current_ip();
+    $criteria_test_flag = true;             // If set to false, nothing will display
+    $temp_attribute_values = array();       // Temporarily stores each value of an attribute (if multiple)
+
+    /* Attribute Conditions */
+	$attributeNames = array(
+		'timezone' => 'location.timeZone', // TODO: this will be a special case, not sure yet how I will treat it
+        'continent' => 'continent',
+        'country' => 'country',
+        'most_specific_subdivision' => 'mostSpecificSubdivision',
+        'region' => 'mostSpecificSubdivision',
+        'state' => 'mostSpecificSubdivision',
+        'city' => 'city',
+	);
+
+	foreach ($attributeNames as $shortcodeParamName => $maxmindName) {
+		if (!empty($atts_array[$shortcodeParamName])) {
+			// ...
+			$actualValues = array();
+			if (isset($info->{$maxmindName}->name)) {
+				$actualValues[] = $info->{$maxmindName}->name;
+			} else if(isset($info->{$maxmindName}->isoCode)) {
+				$actualValues[] = $info->{$maxmindName}->isoCode;
+			} else if(isset($info->{$maxmindName}->code)) {
+				$actualValues[] = $info->{$maxmindName}->code;
+            }
```

Comment:
Since
```
$attributeNames = array(
		'timezone' => 'location.timeZone' ...);
```
I'm not sure if the following would work or not.
```
else if(isset($info->{$maxmindName})) {
				$actualValues[] = $info->{$maxmindName};
            }
```
Or does `$attributeNames`' first element _pointer value_(?) need to be changed to `'location->timeZone'`

Reply:
You're right, this will need a special treatment.